frontend/src/js/external-forms/form-concept-group/FormConceptGroup.tsx

```diff
@@ -196,9 +203,43 @@ const FormConceptGroup = (props: Props) => {
         }
         dropzoneChildren={({ isOver, item }) =>
           isOver && isMovedObject(item)
-            ? t("externalForms.common.concept.copying")
+            ? t("externalForms.common.concept.moving")
             : props.attributeDropzoneText
         }
+        dropBetween={(i: number) => {
+          return (item: DragItemConceptTreeNode) => {
+            if (props.isValidConcept && !props.isValidConcept(item))
+              return null;
+
+            if (isMovedObject(item)) {
+              if (exists(item.dragContext.deleteInOrigin)) {
+                item.dragContext.deleteInOrigin();
```

Comment:
So right now you're
- passing the delete logic in through the drag context
- running the delete logic with the side effect that the values in the form context will have updated
- re-getting the values from the form context
- inserting the value

That's a little complicated. Wouldn't it be possible to just use `props.value` right here, delete the concept or value right here and insert right here as well? So no change to `dragContext`, no call to `getValues`?

Reply:
I agree with you that it is complicated. That were my concerns with this solution as well. However I do not see a different one for the following reason:
If someone drags between two different DropzoneLists then we can not simply delete the element in the target List. The only other solution I can imagine right now (maybe you have a better one in mind), would be to pass the field name where the dragged element originated from. The downside of that would be, that it would increase the code complexity in the deletion logic. 

On the other hand if you want to reduce the complexity and limit the functionality in the sense, that moving between two lists is no longer possible. That could be implemented as well. But I think that should be an internal discussion with the product owners.

Reply:
Hmm. Thought about this again. What I'm trying to avoid is calling `getValues` on the own fieldname. We're passing down `props.value` for the field value. So we should avoid getting the same data from context with getValues. And I'm trying to avoid the implicit side effect of the deletion. 

I'd propose:
- to rename both deleteInForm and deleteInOrigin to be the same, e.g. `deleteFromOtherField` to signal that that's a feature that's ONLY there to support drag-drop moving **across** fields.
- to add a `movedFromFieldName` to the dragContext
- if (isMovedObject(item)), to check if the `dragContext.movedFromFieldName` is the same as the own props.fieldName. If yes, we'd do the delete + insert logic right here, starting from `props.value`, no call to `getValues` anymore. If no, we'd "call & forget" `deleteFromOtherField`, since it won't influence our own props.value. Side effect, yes, but it won't influence our logic here. Then we'd do only the insert logic here.

Does that make sense? Con:
- a little more verbose than your solution right now
- the delete logic will be "in two places"
Pro: 
- it will model more explicitly that "moving from other fields" is a feature we want
- it will avoid getting the same state from props AND context
- we won't have to deal with a possible, implicit side effect